### PR TITLE
Add User-Agent header on HTTP requests and WSS

### PR DESCRIPTION
**Pull request by AlexandrosKyriakakis**
## Context 
- Adding User-Agent header on HTTP requests and WSS
- Similar to [alpaca-py](https://github.com/alpacahq/alpaca-py/blob/695f23de44c5410849b88acb77bdad09c327fb9f/alpaca/common/websocket.py#L78)

### CODE REVIEW

File: alpaca/version.go
Comment (gnvk) [inline alpaca/version.go:3]: Please consider using https://pkg.go.dev/runtime/debug#ReadBuildInfo instead.

Comment (raja) [reply alpaca/version.go:3]: ^ex: ```go
build, ok := debug.ReadBuildInfo()
if !ok {
  log.Printf("Failed to read build info")
  return
}

for _, dep := range build.Deps {
  fmt.Printf("%s %s\n", dep.Path, dep.Version)
}
```

File: marketdata/rest.go
Comment (raja) [inline marketdata/rest.go:92]: why not encode formatted str in version.go?

**Review by raja — CHANGES_REQUESTED** [review]

File: marketdata/stream/conn_nhoory.go
Comment (AlexandrosKyriakakis) [reply marketdata/stream/conn_nhoory.go]: Nice Catch! 
Addressed: https://github.com/alpacahq/alpaca-trade-api-go/pull/254/commits/0a55aca0ed5eac63f4259b7f71116c04385650b8

File: marketdata/rest.go
Comment (AlexandrosKyriakakis) [reply marketdata/rest.go]: Addressed: https://github.com/alpacahq/alpaca-trade-api-go/pull/254/commits/0a55aca0ed5eac63f4259b7f71116c04385650b8

File: alpaca/version.go
Comment (leki75) [inline alpaca/version.go:35]: minor: I suggest following https://www.rfc-editor.org/rfc/rfc9110.html#name-user-agent (or https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent).

Example:
`"ApcaGo/" + moduleVersion + " GoRuntime/" + goVersion`

File: alpaca/version.go
Comment (leki75) [inline alpaca/version.go:28]: minor: I would stick to `runtime.Version()` instead of the build version, I think it is more helpful. Moreover, if build info is not available we can still return the header!

File: alpaca/version.go
Comment (leki75) [inline alpaca/version.go:18]: minor: `init` is evil, but instead of `sync.Once` we can do it as https://github.com/uber-go/guide/blob/master/style.md#avoid-init suggests:
```go
const repo = "github.com/alpacahq/alpaca-trade-api-go"

var version = initVersion()

func initVersion() string {
    moduleVersion := "unknown"
    if buildInfo, ok := debug.ReadBuildInfo(); ok {
		for _, dep := range buildInfo.Deps {
			if strings.HasPrefix(dep.Path, repo) {
				moduleVersion = dep.Version
				break
			}
		}
    }
    return fmt.Sprintf("ApcaGo/%s GoRuntime/%s", moduleVersion, runtime.Version())
}

func Version() string {
    return version
}
```

File: marketdata/stream/conn_nhoory.go
Comment (leki75) [inline marketdata/stream/conn_nhoory.go:28]: minor: While manipulating directly the underlying map is ok now, using the Set or Add methods of `http.Header` has the benefit of canonicalizing the header key.
> Get assumes that all keys are stored in canonical form. To use non-canonical keys, access the map directly.

**Review by leki75 — COMMENTED** [review]
LGTM with minor comments

File: alpaca/version.go
Comment (raja) [inline alpaca/version.go:35]:
```suggestion
		encodedVersions =  fmt."APCA-GO/" + moduleVersion + " GoRuntime/" + goVersion
```

now we are really being nitpicky :)

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent#syntax

ex:
```
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)
Chrome/51.0.2704.103 Safari/537.36
Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0
Mozilla/5.0 (platform; rv:geckoversion) Gecko/geckotrail Firefox/firefoxversion
```

**Comment by raja** [comment]
@AlexandrosKyriakakis going to address reviews?

File: alpaca/version.go
Comment (AlexandrosKyriakakis) [reply alpaca/version.go]: Addressed: https://github.com/alpacahq/alpaca-trade-api-go/pull/254/commits/783440c543b38222ed2b864afaa78626d5f9968e

File: alpaca/version.go
Comment (AlexandrosKyriakakis) [reply alpaca/version.go]: Nice Catch! 🙏🏼 

Addressed: https://github.com/alpacahq/alpaca-trade-api-go/pull/254/commits/689e33c0262c4cc4e0f01fe704c8a2a8ea9248a8

File: marketdata/stream/conn_nhoory.go
Comment (AlexandrosKyriakakis) [reply marketdata/stream/conn_nhoory.go]: Addressed: https://github.com/alpacahq/alpaca-trade-api-go/pull/254/commits/858800b73d759b9614842be2439ff1c9e00a3407

Is this what we want?

File: alpaca/version.go
Comment (AlexandrosKyriakakis) [reply alpaca/version.go]: Addressed: https://github.com/alpacahq/alpaca-trade-api-go/commit/783440c543b38222ed2b864afaa78626d5f9968e

**Review by leki75 — APPROVED** [review]
LGTM

**Comment by AlexandrosKyriakakis** [comment]
Hey @raja I think I need also a re-review from you before merging.

**Review by raja — APPROVED** [review]

**Pull request merged**
